src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -701,6 +701,10 @@ pass::EliminateTranspose::EliminateTranspose() {
         }
 
         const auto& order_values = order_const->cast_vector<int64_t>();
+        // Cannot eliminate Transpose when empty order_value.
+        if (order_values.size() == 0)
+            return false;
```

Comment:
please, submit all changes, which are not in CPU plugin, as separate PRs with appropriate tests

Reply:
This issue is triggered when put preprocess before transformation for legacy ov api + cpu plugin api 2.0

Reply:
Independently on why these changes are needed, changes in quite different components should be submitted as a separate PRs. They will be reviewed by different people.

Reply:
Got it, will split such changes into multiple separated PRs per components.

Reply:
Split transformation issue into this PR: https://github.com/openvinotoolkit/openvino/pull/18130